Checkout: shows website URL instead of theme slug as item description

alleh opened the pull request:
Suggestion for #2420

lancewillett commented:
@gziolo Can you take a quick look and review?

gziolo commented on client/my-sites/upgrades/cart/cart-item.jsx:
```diff
@@ -80,7 +80,7 @@ module.exports = React.createClass( {
 			info = getIncludedDomain( this.props.cartItem );
 		} else {
-			info = domain;
+		    info = this.props.cartItem.is_domain_registration ? domain : this.props.selectedSite.domain;
```
@alleh: Many thanks for looking at this issue :)

We have several conditions here and it would be great to avoid nested conditions here like:
```js
} else if ( this.props.cartItem.is_domain_registration ) {
	info = domain;
} else {
	info = this.props.selectedSite.domain;
}  
```

BTW, we indent with tabs:
> Indentation with tabs.

This is from our [JavaScript Coding Guidelines](https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/javascript.md). You can check this document to find more details.

gziolo commented:
I tested this change and it looks like we change to much with this code. To give an example:
- you have more than one domain assigned to the site
- there is a domain without privacy protection and it isn't set as a primary domain
- when you try to add privacy protection it shows primary domain name in the cart instead of the domain you are about to modify

Taking that into account we should make sure we display primary domain name only for themes. So the condition presented above should change to sth like:

```js
} else if ( isTheme( this.props.cartItem ) ) {
	info = this.props.selectedSite.domain;
} else {
	info = domain;
}  
```

alleh commented:
@gziolo Good points! Did what you said, everything seems to work as expected now!

![checkout-wp-calypso](https://cloud.githubusercontent.com/assets/3101006/12432726/376184da-befd-11e5-9cf3-ab3deb742616.png)

This looks ok right?
Btw, can I rebase my commits to one or will it mess up something? 

"This branch has conflicts that must be resolved" must be because of I changed to tabs and removed whitespace? (Sorry I'm new to PRs)

gziolo commented:
@alleh Great job on this one!

To answer your questions.

> Btw, can I rebase my commits to one or will it mess up something?

It's good practice to squash minor commits. There is contributing guide which contains [Lifecycle of a Pull Request](https://github.com/Automattic/wp-calypso/blob/master/CONTRIBUTING.md#lifecycle-of-a-pull-request) section. There are two bullet points that are related to your question:
> - Don’t be afraid to change, [squash](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html), and rearrange commits or to force push - `git push -f origin fix/something-broken`. Keep in mind, however, that if other people are committing on the same branch then you can mess up their history. You are perfectly safe if you are the only one pushing commits to that branch.
- Squash minor commits such as typo fixes or [fixes to previous commits](http://fle.github.io/git-tip-keep-your-branch-clean-with-fixup-and-autosquash.html) in the pull request.

> "This branch has conflicts that must be resolved" must be because of I changed to tabs and removed whitespace? (Sorry I'm new to PRs)

You need to rebase your branch with upstream master to resolve conflicts. I recommend a document which describes how we do it, in particular section
[Keeping Your Branch Up To Date](https://github.com/Automattic/wp-calypso/blob/master/docs/git-workflow.md#short-branches-merge-early-and-often).

Ping me when branch is updated, I'll retest and merge with master branch :)

alleh commented:
@gziolo Should be ready to go now! Thanks for the help :)

gziolo commented:
It looks and works as expected! Many thanks for your contribution.
I'm going to merge and deploy it now :)

lancewillett commented:
Good work alleh and gziolo. :)
